src/Uno.UI/UI/Xaml/DependencyPropertyDetails.cs

```diff
@@ -130,6 +130,8 @@ internal DependencyPropertyDetails(DependencyProperty property, Type dependencyO
 		/// <param name="precedence">The precedence level to set the value at</param>
 		internal void SetValue(object? value, DependencyPropertyValuePrecedences precedence)
 		{
+			Property.ValidateValue(value);
```

Comment:
That seems like a very hot path. Is it done the same way in winui?

Reply:
@jeromelaban It's quite similar. Every call to SetValue is going to be validated. I'd guess this call would be negligible compared to the other work done by SetValue